### PR TITLE
Convert StarkNet -> Starknet

### LINKED ISSUE #102: Convert StarkNet -> Starknet

### PR DESCRIPTION
**Pull request by franciszekjob**
## Describe your changes

- update `StarkNet` to `Starknet` in typed data docs
- update `StarkNet` to `Starknet`  in readme

## Linked issues

Closes #102 
## Breaking changes

- [ ] This issue contains breaking changes

### CODE REVIEW

File: Sources/Starknet/Data/TypedData/StarknetTypedData.swift [hunk 1]
```diff
@@ -384,7 +384,7 @@ public struct StarknetTypedData: Codable, Equatable, Hashable {
 
     public func getMessageHash(accountAddress: Felt) throws -> Felt {
         try hashArray([
-            Felt.fromShortString("StarkNet Message")!,
+            Felt.fromShortString("Starknet Message")!,
```
Comment (DelevoXDG) [inline Sources/Starknet/Data/TypedData/StarknetTypedData.swift:387]: > The `PREFIX_MESSAGE` **must be** `StarkNet Message`.

per the [SNIP](https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-12.md#prefix-message). 

Let's verify all changes to `StarkNet` spelling in typed data and make sure we should actually make them

File: Tests/StarknetTests/Data/TypedDataTests.swift [hunk 2]
```diff
@@ -27,7 +27,7 @@ final class TypedDataTests: XCTestCase {
     }
 
     static let domainTypeV0 = (
-        "StarkNetDomain",
+        "StarknetDomain",
```
Comment (DelevoXDG) [inline Tests/StarknetTests/Data/TypedDataTests.swift:30]: This also shouldn't be the case per SNIP-12.

File: Tests/StarknetTests/Data/TypedDataTests.swift [hunk 3]
```diff
@@ -396,57 +396,57 @@ final class TypedDataTests: XCTestCase {
...
                 Self.CasesRev0.tdValidate,
                 "0xcd2a3d9f938e13cd947ec05abc7fe734df8dd826",
-                "0x6038f35de58f40a6afa9d359859b2f930e5eb987580ba6875324cc4dbfcee"
+                "0x9410650b3964284ea17d69e509c8d9c17db0e7493b0d966cbd0cdeafca0160"
             ),
             (
                 Self.CasesRev1.td,
                 "0xcd2a3d9f938e13cd947ec05abc7fe734df8dd826",
-                "0x7f6e8c3d8965b5535f5cc68f837c04e3bbe568535b71aa6c621ddfb188932b8"
+                "0x68bc218000ceaa6c38ab8adeeac21c339e807bab00a26dd13604fa98d867ec"
```
Comment (DelevoXDG) [inline Tests/StarknetTests/Data/TypedDataTests.swift:429]: Hashes shouldn't be changed as a result of this change.

File: Tests/StarknetTests/Resources/Accounts/starknet_open_zeppelin_accounts.json [hunk 4]
```diff
@@ -17,4 +17,4 @@
       "salt": "0x1a5dd1281a0f2a74"
     }
   }
-}
+}
```
Comment (DelevoXDG) [inline Tests/StarknetTests/Resources/Accounts/starknet_open_zeppelin_accounts.json:20]: Why?

File: Tests/StarknetTests/Resources/TypedData/typed_data_rev_0_example.json [hunk 5]
```diff
@@ -1,23 +1,47 @@
 {
   "types": {
-    "StarkNetDomain": [
-      { "name": "name", "type": "felt" },
-      { "name": "version", "type": "felt" },
-      { "name": "chainId", "type": "felt" }
+    "StarknetDomain": [
```
Comment (DelevoXDG) [inline Tests/StarknetTests/Resources/TypedData/typed_data_rev_0_example.json:3]: Imo we shouldn't format these: it doesn't make those more readable and is unrelated to this PR.

Comment (franciszekjob) [reply Tests/StarknetTests/Resources/TypedData/typed_data_rev_0_example.json]: I've removed unnecessary formatting changes.

File: Sources/Starknet/Providers/StarknetProviderProtocol.swift [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import Foundation
 
-/// Provider used to interact with the StakNet blockchain.
+/// Provider used to interact with the StarkNet blockchain.
```
Comment (DelevoXDG) [inline Sources/Starknet/Providers/StarknetProviderProtocol.swift:3]: Let's also 🤭 
```suggestion
/// Provider used to interact with the Starknet blockchain.
```

**Review by DelevoXDG — CHANGES_REQUESTED** [review]

**Review by ddoktorski — APPROVED** [review]

**Review by DelevoXDG — APPROVED** [review]

**Pull request merged**